conans/test/integration/hooks/attribute_checker_test.py

```diff
@@ -0,0 +1,54 @@
+# coding=utf-8
+
+import os
+import shutil
+import tempfile
+import textwrap
+import unittest
+
+from parameterized import parameterized
+
+from conans.client.graph.python_requires import ConanPythonRequire
+from conans.client.hook_manager import HookManager
+from conans.client.loader import ConanFileLoader, ProcessedProfile
+from conans.test.utils.tools import TestBufferConanOutput, save
+
+
+class AttributeCheckerTest(unittest.TestCase):
+    conanfile_base = textwrap.dedent("""\
+        from conans import ConanFile
+
+        class AConan(ConanFile):
+            {placeholder}
+        """)
+    conanfile_basic = conanfile_base.format(placeholder='pass')
+    conanfile_alias = conanfile_base.format(placeholder='alias = "something"')
+
+    def run(self, *args, **kwargs):
+        hooks_folder = tempfile.mkdtemp()
```

Comment:
I think we have util for this: `temp_folder()`

Reply:
`temp_folder()` leaves trash in my computer, I don't like it 🗑 

I would go for the contextmanger in the `TestCase.run` method, I really like that pattern.

Reply:
touché 😅 ! ... I'll change it.